### PR TITLE
Fixing the OLED update screen values

### PR DESCRIPTION
**Pull request by brandonrc**
Adding extern TxConfig object into the devOLED lib so we can access the helper functions for GetTelm, GetPower, and GetRate. 

Updated the OLED lib to adjust the packet rate enum based on frequency.

### CODE REVIEW

File: src/lib/OLED/devOLED.cpp [hunk 1]
```diff
@@ -25,9 +28,9 @@ static int event()
     {
         return DURATION_IGNORE; // we are still displaying the startup message, so don't change the timeout
     }
-    OLED.updateScreen(OLED.getPowerString((PowerLevels_e)POWERMGNT::currPower()),
-                        OLED.getRateString((expresslrs_RFrates_e)ExpressLRS_currAirRate_Modparams->enum_rate),
-                        OLED.getTLMRatioString((expresslrs_tlm_ratio_e)(ExpressLRS_currAirRate_Modparams->TLMinterval)), thisCommit);
+    OLED.updateScreen(OLED.getPowerString((PowerLevels_e)(config.GetPower())),
+                        OLED.getRateString((expresslrs_RFrates_e)(config.GetRate())),
+                        OLED.getTLMRatioString((expresslrs_tlm_ratio_e)(config.GetTlm())), thisCommit);
```
Comment (pkendall64) [inline src/lib/OLED/devOLED.cpp:33]: Should be casting them to the enums if we're just using them as int's in the functions.

**Comment by CapnBry** [comment]
Wouldn't it make more sense to have devOLED have a timeout function that checks the three things it displays and if any of them changes, call `OLED.updateScreen()`?

And probably not the time to discuss this since we're in RC, but why does the **caller** pass 3 strings to updateOled when all three strings are generated by OLED? Just the values can be passed and the draw routine can get the appropriate string.

**Comment by pkendall64** [comment]
> Wouldn't it make more sense to have devOLED have a timeout function that checks the three things it displays and if any of them changes, call `OLED.updateScreen()`?
> 
> And probably not the time to discuss this since we're in RC, but why does the **caller** pass 3 strings to updateOled when all three strings are generated by OLED? Just the values can be passed and the draw routine can get the appropriate string.

The event function is called whenever the config changes so no need to do things on a timeout basis.

**Comment by CapnBry** [comment]
> The event function is called whenever the config changes so no need to do things on a timeout basis.

Right, but the event function is not called when the current value changes, just when the config changes.  Who cares what the configuration is when that doesn't change?

Why use a dynamic medium (live OLED display) to display something static? Someone could just print a sticker that says what the config is and stick it on! :-P

**Comment by pkendall64** [comment]
> > The event function is called whenever the config changes so no need to do things on a timeout basis.
> 
> Right, but the event function is not called when the current value changes (POWERMGNT::currPower, TLMinterval, enum_rate), just when the config changes. Who cares what the configuration is when that doesn't change?
> 
> Why use a dynamic medium (live OLED display) to display something static? Someone could just print a sticker that says what the config is and stick it on! :-P

Oh you mean by leaving the original code and just use a timer to update the screen, for dynamic power etc. That makes sense now.

**Comment by CapnBry** [comment]
> Oh you mean by leaving the original code and just use a timer to update the screen, for dynamic power etc. That makes sense now.

Yup, just remove the event and run it in a timeout. Check the 3 things on the screen against a static lastval. Any of them changed? OLED.updateDisplay.

**Comment by CapnBry** [comment]
Like this. I can push this if @brandonrc agrees. It updates dynamically every 300ms if there's a change in the values and is more useful than just seeing your config.
```
diff --git a/src/lib/OLED/devOLED.cpp b/src/lib/OLED/devOLED.cpp
index 76281552..dd4421e7 100644
--- a/src/lib/OLED/devOLED.cpp
+++ b/src/lib/OLED/devOLED.cpp
@@ -10,37 +10,38 @@
 class OLED OLED;

 static const char thisCommit[] = {LATEST_COMMIT, 0};
-static bool startup = false;

 static int start()
 {
     OLED.displayLogo();
-    startup = true;
     return 1000;    // set callback in 1s
 }

-static int event()
+static int timeout()
 {
-    if (startup)
+    static PowerLevels_e lastPower;
+    static expresslrs_RFrates_e lastRate;
+    static expresslrs_tlm_ratio_e lastRatio;
+
+    PowerLevels_e newPower = POWERMGNT::currPower();
+    expresslrs_RFrates_e newRate = ExpressLRS_currAirRate_Modparams->enum_rate;
+    expresslrs_tlm_ratio_e newRatio = ExpressLRS_currAirRate_Modparams->TLMinterval;
+
+    if (lastPower != newPower || lastRate != newRate || lastRatio != newRatio)
     {
-        return DURATION_IGNORE; // we are still displaying the startup message, so don't change the timeout
+        OLED.updateScreen(newPower, newRate, newRatio, thisCommit);
+        lastPower = newPower;
+        lastRate = newRate;
+        lastRatio = newRatio;
     }
-    OLED.updateScreen(OLED.getPowerString((PowerLevels_e)POWERMGNT::currPower()),
-                        OLED.getRateString((expresslrs_RFrates_e)ExpressLRS_currAirRate_Modparams->enum_rate),
-                        OLED.getTLMRatioString((expresslrs_tlm_ratio_e)(ExpressLRS_currAirRate_Modparams->TLMinterval)), thisCommit);
-    return DURATION_NEVER;
-}

-static int timeout()
-{
-    startup = false;
-    return event();
+    return 300; // check for updates every 300ms
 }

 device_t OLED_device = {
-    .initialize = NULL,
+    .initialize = nullptr,
     .start = start,
-    .event = event,
+    .event = nullptr,
     .timeout = timeout
 };
```

That is kinda hard to read as a diff so here it is as code:
image

**Review by pkendall64 — APPROVED** [review]

**Pull request merged**